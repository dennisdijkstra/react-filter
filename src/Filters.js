import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Filters extends Component {
    static propTypes = {
        updateSearchValue: PropTypes.func.isRequired,
        updateSelectValue: PropTypes.func.isRequired,
        search: PropTypes.string.isRequired,
        types: PropTypes.arrayOf(PropTypes.string).isRequired,
    };

    getSearchInput = (event) => {
        const { updateSearchValue } = this.props;
        const input = event.target.value;

        updateSearchValue(input);
    }

    getSelectInput = (event) => {
        const { updateSelectValue } = this.props;
        const input = event.target.value;

        updateSelectValue(input);
    }

    render() {
        const { search, types } = this.props;
        return (
            <div className="filter">
                <h2 className="filter-title">Filters</h2>
                <div className="filter-input">
                    <p className="filter-input-title">Search:</p>
                    <input type="text" onChange={this.getSearchInput} value={search} />
                </div>
                <div className="filter-input">
                    <p className="filter-input-title">Type of object:</p>
                    <select name="type" onChange={this.getSelectInput}>
                        {types.map(type => (
                            <option value={type}>{type.charAt(0).toUpperCase() + type.slice(1)}</option>
                        ))}
                    </select>
                </div>
            </div>
        );
    }
}

export default Filters;
