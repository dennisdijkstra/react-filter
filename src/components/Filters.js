import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Filters extends Component {
    static propTypes = {
        updateStateValues: PropTypes.func.isRequired,
        search: PropTypes.string.isRequired,
        select: PropTypes.string.isRequired,
        types: PropTypes.arrayOf(PropTypes.string).isRequired,
    };

    getInput = () => {
        const { updateStateValues } = this.props;
        const search = this.search.value;
        const select = this.select.value;

        updateStateValues(search, select);
    }

    render() {
        const { search, select, types } = this.props;

        return (
            <div className="filter">
                <h2 className="filter-title">Filters</h2>
                <div className="filter-input">
                    <p className="filter-input-title">Search:</p>
                    <input type="text" onChange={this.getInput} value={search} ref={(input => this.search = input)} />
                </div>
                <div className="filter-input">
                    <p className="filter-input-title">Type of object:</p>
                    <select name="type" onChange={this.getInput} value={select} ref={(input => this.select = input)}>
                        <option value="all">All</option>
                        {types.map(type => (
                            <option value={type}>{type.charAt(0).toUpperCase() + type.slice(1)}</option>
                        ))}
                    </select>
                </div>
            </div>
        );
    }
}

export default Filters;
