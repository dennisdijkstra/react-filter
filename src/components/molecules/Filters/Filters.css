.filter {
    width: 200px;
    min-height: 100%;
    padding: 30px;
    color: #fff;
    background-color: #000;

    &-input {
        margin-bottom: 24px;
    }

    &-title {
        margin-bottom: 24px;
        font-size: 18px;
        font-weight: 400;
    }

    &-input-title {
        margin-bottom: 8px;
        font-size: 14px;
        font-weight: 100;
    }
}
