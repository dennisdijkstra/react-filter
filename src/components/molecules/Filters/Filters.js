import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import s from './Filters.css';

const Filters = (props) => {
    const { selectCategories, initialValues, filter } = props;
    const { search, select } = initialValues;

    const filterItems = () => {
        setTimeout(() => filter(), 0);
    };

    return (
        <div className={s.filter}>
            <h2 className={s['filter-title']}>Filters</h2>
            <form>
                <div className={s['filter-input']}>
                    <p className={s['filter-input-title']}>Search:</p>
                    <Field
                        name="search"
                        component="input"
                        value={search}
                        type="text"
                        onChange={filterItems}
                    />
                </div>
                <div className={s['filter-input']}>
                    <p className={s['filter-input-title']}>Type of object:</p>
                    <Field name="select" component="select" value={select} onChange={filterItems}>
                        <option value="all">All</option>
                        {selectCategories.map(category => (
                            <option
                                key={category.id}
                                value={category.type}
                            >
                                {category.type.charAt(0).toUpperCase() + category.type.slice(1)}
                            </option>
                        ))}
                    </Field>
                </div>
            </form>
        </div>
    );
};

export default reduxForm({
    form: 'filters',
})(Filters);

Filters.propTypes = {
    selectCategories: PropTypes.arrayOf(PropTypes.object).isRequired,
    filter: PropTypes.func.isRequired,
    initialValues: PropTypes.shape({
        search: PropTypes.string.isRequired,
        select: PropTypes.string.isRequired,
    }).isRequired,
};
