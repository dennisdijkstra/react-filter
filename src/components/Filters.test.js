import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Filters from './Filters';

jest.mock('./Filters');
configure({ adapter: new Adapter() });

const Filter = new Filters();
console.log(Filter);

beforeEach(() => {
    Filters.mockClear();
});

describe('Filters', () => {
    it('should be defined', () => {
        expect(Filters).toBeDefined();
    });

    it('should render correctly', () => {
        const selectCategories = [{ id: '3445', type: 'all' }];
        const component = shallow(
            <Filters selectCategories={selectCategories} />,
        );
        expect(component).toMatchSnapshot();
    });

    it('should be defined', () => {
        expect(Filter.getInput).toBeDefined();
    });
});
