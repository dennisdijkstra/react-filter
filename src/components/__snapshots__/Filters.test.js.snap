// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    selectCategories={
      Array [
        Object {
          "id": "3445",
          "type": "all",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
